app/static/admin/js/pages/gen_form/form-donor.js

```diff
@@ -1,139 +1,165 @@
-    //   var value = (ui && ui.item && ui.item.value) || e.target.value;
-    //
-    //   if (!value || value == "") {
-    //     setDonorForm.apply(null, null);
-    //     return;
-    //   }
-    //
-    //   if (util.check("name", value)) return;
-    //
-    //   console.log(value);
-    //
-    //   $.ajax({
-    //     type: "GET",
-    //     url: "/api/donor",
-    //     dataType: "json",
-    //     data: {
-    //       donor_id: 127 // TODO: Change to real donor_id
-    //     },
-    //     success: function() {
-    //       setDonorForm.apply(null, arguments);
-    //     },
-    //     error: function() {
-    //       console.error(arguments);
-    //     }
-    //   });
-    // }.bind(this);
-
-    var getDonorInfo = function(e, ui) {
-      var nameId = (ui && ui.item && ui.item.value) || e.target.value;
-
-      if (!nameId || nameId == "") {
-        setDonorForm.apply(null, null);
-        return;
-      }
-
-      if (util.check("name", nameId)) return;
-
-      setDonorForm(store[nameId]);
+    var getDonorInfo = function (e, ui) {
+        var nameId = (ui && ui.item && ui.item.value) || e.target.value;
+
+        if (!nameId || nameId == "") {
+            setDonorForm.apply(null, null);
+            return;
+        }
+
+        if (util.check("name", nameId)) return;
+
+        setDonorForm(store[nameId]);
 
     }.bind(this);
 
-    var setDonorForm = function(data) {
-      if (!data) {
-        util.emptyAllFields(this.input, [this.input.name]);
-        util.setButton(this.button, "new");
-
-        // TODO
-        donation.printDonationList([]);
-        return;
-      }
-
-      this.input.email.value = data.email;
-      this.input.telephone.value = data.telephone_number;
-      this.input.mobile.value = data.mobile_number;
-      this.input.ref.value = data.customer_ref;
-      this.input.needReceipt.value = data.want_receipt;
-      this.input.address.value = data.address_line;
-      this.input.city.value = data.city;
-      this.input.province.value = data.province;
-      this.input.postalCode.value = data.postal_code;
-
-      util.setButton(this.button, "existing");
-
-
-      // TODO
-      // $.ajax({
-      //   type: "GET",
-      //   url: "/api/donation_list",
-      //   dataType: "json",
-      //   data: {
-      //     donor_id: data.id // TODO: Change to real donor_id
-      //   },
-      //   success: function() {
-      //     setDonorForm.apply(null, arguments);
-      //   },
-      //   error: function() {
-      //     console.error(arguments);
-      //   }
-      // });
-      //
-      // donation.printDonationList(data.donation_records);
-    }.bind(this.dom);
+    var setDonorForm = function (data) {
+        if (!data) {
+            util.emptyAllFields(this.input, [this.input.name]);
+            util.setButton(this.button, "new");
+
+            donation.getDonation(null);
+            return;
+        }
+
+        this.input.name.value = data.donor_name;
```

Comment:
Perhaps these could be refactored into a function that takes two objects (input and data) and set the value of input to match data? This does mean that you need to change your variable names but it will be definitely more scalable and less typing.

Reply:
I see where you are coming from. That would be ideal if naming conventions are defined and enforced.